src/packages/media/media/components/input-image-cropper/image-cropper-focus-setter.element.ts

```diff
@@ -79,73 +93,145 @@ export class UmbImageCropperFocusSetterElement extends LitElement {
 				this.imageElement.style.height = '100%';
 			}
 
+			this.#resetCoords();
+			
 			this.imageElement.style.aspectRatio = `${imageAspectRatio}`;
 			this.wrapperElement.style.aspectRatio = `${imageAspectRatio}`;
 		};
+	}
 
-		if (!this.hideFocalPoint) {
-			this.#addEventListeners();
+	#onFocalPointUpdated() {
+		if (this.#isCentered(this.#focalPoint)) {
+			this.#resetCoords();
 		}
 	}
 
-	async #addEventListeners() {
-		await this.updateComplete; // Wait for the @query to be resolved
-		this.imageElement?.addEventListener('mousedown', this.#onStartDrag);
-		window.addEventListener('mouseup', this.#onEndDrag);
+	#coordsToFactor(x: number, y: number) {
+		const top = (y / 100) / y * 50;
+		const left = (x / 100) / x * 50;
+
+		return { top, left }; 
 	}
 
-	#removeEventListeners() {
-		this.imageElement?.removeEventListener('mousedown', this.#onStartDrag);
-		window.removeEventListener('mouseup', this.#onEndDrag);
+	#setFocalPoint(x: number, y: number, width: number, height: number) {
+
+		const left = clamp((x / width), 0, 1);
+		const top = clamp((y / height), 0, 1);
+
+		this.#coordsToFactor(x, y);
+
+		this.dispatchEvent(
+			new CustomEvent('change', {
+				detail: { left, top },
+				bubbles: false,
+				composed: false,
+			}),
```

Comment:
Consider creating a custom event class for this to stay in line with other custom events.

Reply:
Yeah, this was just how it already works, so didn't change that part. But a custom event class is probably better.

Reply:
@iOvergaard should the custom event be added here?
https://github.com/umbraco/Umbraco.CMS.Backoffice/tree/97f256a195d8301bcc68ba21ed879240cd0d663d/src/packages/core/event

Reply:
Nah, I would put it in the components/input-image-cropper folder as well and call it something like `UmbImageCropChange`. From there you can export it through the index.ts file.

Also be sure to add the crops directly to the event class rather than use `detail`.

Reply:
I called it `UmbFocalPointChangeEvent` as it trigget when focal point is set (click or drag):
https://github.com/umbraco/Umbraco.CMS.Backoffice/pull/1021/commits/f1449ddab26802a46750e4e101e78b7cdca41e78

not which event this one is?

![image](https://github.com/user-attachments/assets/4c45307c-fcf0-4b2b-bcc5-2a8eb5163ac2)

Do we want `top` and `left` properties like in `UmbProgressEvent`?
https://github.com/umbraco/Umbraco.CMS.Backoffice/blob/97f256a195d8301bcc68ba21ed879240cd0d663d/src/packages/core/event/progress.event.ts#L7

Perhaps there is an type containing `top` and `left` properties? and if it anytime later need additional properties?

Reply:
There is the `UmbFocalPointModel` which would be suitable to use.

Reply:
I replaced the custom event in image crop as well with `UmbImageCropChangeEvent`.

So we have these two:

- `UmbImageCropChangeEvent`
- `UmbFocalPointChangeEvent`